functests/dpdk/dpdk.go

```diff
@@ -0,0 +1,117 @@
+package dpdk
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+)
+
+// Entry to find in the pod logfile
+const logEntry = "Accumulated forward statistics for all ports"
+
+var testDpdkNamespace string
+
+func init() {
+	testDpdkNamespace = os.Getenv("DPDK_TEST_NAMESPACE")
+	if testDpdkNamespace == "" {
+		testDpdkNamespace = "dpdk-testing"
+	}
+}
+
+var _ = Describe("dpdk", func() {
+	var _ = Context("Run a sanity test on a worker", func() {
+		It("Should forward and receive packets", func() {
+			var out string
+			p := findPod()
+			pods.WaitForCondition(client.Client, p, corev1.ContainersReady, corev1.ConditionTrue, 2*time.Minute)
+			By("Parsing output from the DPDK application")
+			Eventually(func() string {
+				out = getPodLog(p)
+				return out
+			}, 8*time.Minute, 1*time.Second).Should(ContainSubstring(logEntry),
+				"Cannot find accumulated statistics")
+			checkRxTx(out)
+		})
+	})
+})
+
+// checkRxTx parses the output from the DPDK test application
+// and verifies that packets have passed the NIC TX and RX queues
+func checkRxTx(out string) {
+	str := strings.Split(out, "\n")
+	for i := 0; i < len(str); i++ {
+		if strings.Contains(str[i], logEntry) {
+			i++
+			d := getNumberOfPackets(str, i)
+			Expect(d).Should(BeNumerically(">", 0), "number of received packets should be greater than 0")
+
+			i++
+			d = getNumberOfPackets(str, i)
+			Expect(d).Should(BeNumerically(">", 0), "number of transferred packets should be greater than 0")
+		}
```

Comment:
Can we break the for if we enter the if section?

Reply:
@SchSeba, since the required lines will be at the end of the log, a break is not so important here. But I can add it, in any case.